replay/replay.go

```diff
@@ -576,17 +598,28 @@ const (
 // database so that the replay runner has access to internal Pebble events.
 func (r *Runner) eventListener() pebble.EventListener {
 	var writeStallBegin time.Time
+	var writeStallReason string
 	l := pebble.EventListener{
 		BackgroundError: func(err error) {
 			r.err.Store(err)
 			r.cancel()
 		},
-		WriteStallBegin: func(pebble.WriteStallBeginInfo) {
-			r.metrics.writeStalls.Add(1)
+		WriteStallBegin: func(info pebble.WriteStallBeginInfo) {
+			r.writeStallMetrics.Lock()
+			defer r.writeStallMetrics.Unlock()
+			writeStallReason = info.Reason
+			// Take just the first word of the reason.
+			if j := strings.IndexByte(writeStallReason, ' '); j != -1 {
```

Comment:
nit: could be worth enumerating the write stall reasons to future proof this from folks revising or adding the write stall reason strings